mbbsd/register.c

```diff
@@ -1162,11 +1179,19 @@ check_register(void)
 
     // �T�{�@�w�n���p���H�c
     if (!is_valid_email(cuser.email)) {
-        more("etc/ensurecontactemail", NA);
+        more("etc/ensurevalidcontactemail", NA);
         pressanykey();
 
         // additional parentheses around "ret =" to please compiler
-        while( (ret = change_contact_email()) );
+        while ((ret = change_contact_email(0)));
+#ifdef FORCE_UPDATE_CONTACT_EMAIL_LASTLOGIN
+    } else if (cuser.lastlogin < FORCE_UPDATE_CONTACT_EMAIL_LASTLOGIN) {
```

Comment:
`cuser.lastlogin` is already updated [here in `mbbsd.c:user_login()`](https://github.com/ptt/pttbbs/blob/master/mbbsd/mbbsd.c#L1197) before `check_register()` is called in the same function.

Instead, you should use `last_login_time`, which is a backup of `cuser.lastlogin` saved early in `start_client()` before `user_login()` is called.

Reply:
It looks like [here in `mbbsd.c:user_login()`](https://github.com/ptt/pttbbs/blob/master/mbbsd/mbbsd.c#L1197)
actually only sets the human-readable format of `cuser.lastlogin` to `lasttime`, but not changes `cuser.lastlogin`
([https://github.com/ptt/pttbbs/blob/master/common/sys/time.c#L153](https://github.com/ptt/pttbbs/blob/master/common/sys/time.c#L153))

However, it looks like [the convention in the code is using `last_login_time`](https://github.com/ptt/pttbbs/blob/master/mbbsd/mbbsd.c#L1269)

I'll follow the convention accordingly.